data/datasource.go

```diff
@@ -371,6 +388,13 @@ func readStdin(source *Source, args ...string) ([]byte, error) {
 	return b, nil
 }
 
+func readPlugin(source *Source, args ...string) ([]byte, error) {
+	if source.gf == nil {
+		return nil, errors.Errorf("Datasource with %s is a plugin datasource but its plugin's get method is nil.", source.Alias)
+	}
+	return source.gf.(func(url *url.URL, headers http.Header, args []string) ([]byte, error))(source.URL, source.header, args)
```

Comment:
we should comment on what is going on here.

Reply:
added a description